step/step.go

```diff
@@ -27,7 +38,21 @@ type DockerBuildPushStep struct {
 	envRepo        env.Repository
 }
 
-func New(logger log.Logger, inputParser stepconf.InputParser, commandFactory command.Factory, pathChecker pathutil.PathChecker, pathProvider pathutil.PathProvider, pathModifier pathutil.PathModifier, envRepo env.Repository) DockerBuildPushStep {
+const (
+	dockerCacheKeyTemplate     = "docker-%s-{{ .OS }}-{{ .Arch }}-{{ .Branch }}-{{ .CommitHash }}"
```

Comment:
Shouldn't a checksum be part of the key (probably the checksum of the Dockerfile)?

Reply:
With dockerfiles that does not work reliably as the image changes often based on the content not the dockerfile. So if any of the code changes and you have COPY command that would result in a new image.

Reply:
Yeah, makes sense. We would probably need to checksum every single file inside the context folder, but this seems a bit overkill.